kube-client/src/api/core_methods.rs

```diff
@@ -35,6 +41,34 @@ where
         self.client.request::<K>(req).await
     }
 
+    /// Get metadata (to-fill-in-later)
+    pub async fn get_metadata(&self, name: &str) -> Result<PartialObjectMeta> {
+        let mut req = self.request.get_metadata(name).map_err(Error::BuildRequest)?;
+        req.extensions_mut().insert("get");
+        self.client.request::<PartialObjectMeta>(req).await
+    }
+
+    /// List metadata (to-fill-in-later)
+    pub async fn list_metadata(&self, lp: &ListParams) -> Result<PartialObjectMetaList> {
+        let mut req = self.request.list_metadata(lp).map_err(Error::BuildRequest)?;
+        req.extensions_mut().insert("list");
+        self.client.request::<PartialObjectMetaList>(req).await
+    }
+
+    /// Watch meta
+    pub async fn watch_metadata(
+        &self,
+        lp: &ListParams,
+        version: &str,
+    ) -> Result<impl Stream<Item = Result<WatchEvent<PartialObjectMeta>>>> {
+        let mut req = self
+            .request
+            .watch_metadata(lp, version)
+            .map_err(Error::BuildRequest)?;
+        req.extensions_mut().insert("watch");
+        self.client.request_events::<PartialObjectMeta>(req).await
```

Comment:
if this works, then this is perfect; a mirror of what the api does (what the crate is meant to do).


..it will present some extra logic to `watcher` to actually set up a persistent watch stream (watcher assumes the input type is the same as the output type, and now we need to go through conversion first), but hopefully we can do something similar to `watch_object` and specialize for that down the line (maybe some `Into<K>` bounds can do it).

Reply:
By persistent here we mean watches that do not terminate on errors? I have to be honest, haven't spent a lot of time looking and understanding how they're implemented in `kube-rs`. I looked at `watch_object`, think it might be work with some bounds. Happy to take a stab at it once this wraps up. Will try to do it before a release is out 🤞🏻

Reply:
Yeah, most informer-style(watcher)/reflector-style/controller based workloads will want to use an infinite watch that retries on de-syncs. Regular `Api::watch` is basically awful for real-world use-cases because of how temperamental the api is, so we forward people to `watcher` or the more specific `watch_object`.

> Happy to take a stab at it once this wraps up. Will try to do it before a release is out 🤞🏻

That would be great ❤️